File: Seccion 18 - Clases en C++/Seccion_18_Python/04_ejercicio_04.py

class Vehiculo:
    
    def __init__(self, marca="null", modelo="null", precio=0.0):
        self.marca = marca
        self.modelo = modelo
        self.precio = precio
    
    def get_precio(self):
        return self.precio
    
    def mostrar_datos(self):
        print("Marca:", self.marca)
        print("Modelo:", self.modelo)
        print("Precio:", self.precio)

    @staticmethod
    def indice_mas_barato(coches: list, n: int):
        indice: int = 0
        precio: float = coches[0].get_precio()
        
        for i in range(1, n):
            if coches[i].get_precio() < precio:
                precio = coches[i].get_precio()
                indice = i
        
        return indice

def main():
    
    coches = []
    numVehiculos = int(input("Digite el número de vehículos: "))
    
    for i in range(numVehiculos):
        print("\nDigite los datos del vehículo {}:".format(i + 1))
        marca = input("Marca: ")
        modelo = input("Modelo: ")
        precio = float(input("Precio: "))
        
        coches.append(Vehiculo(marca=marca, modelo=modelo, precio=precio))
    
    cocheBarato = Vehiculo.indice_mas_barato(coches, numVehiculos)
    
    print("\nEl vehículo más barato es:")
    coches[cocheBarato].mostrar_datos()

if __name__ == "__main__":
    main()

File: Seccion 18 - Clases en C++/Seccion_18_Dart/04_ejercicio_04.dart

import 'dart:io';

class Vehiculo {

  String marca, modelo;
  double precio;

  Vehiculo();
  Vehiculo.datos(this.marca, this.modelo, this.precio);

  double getPrecio() {
    return precio;
  }

  void mostrarDatos() {
    print("Marca: $marca");
    print("Modelo: $modelo");
    print("Precio: $precio");
  }

  static int indiceMasBarato(List<Vehiculo> coches, int n) {
    int indice = 0;
    double precio = coches[0].getPrecio();

    for (int i = 1; i < n; i++) {
      if (coches[i].getPrecio() < precio) {
        precio = coches[i].getPrecio();
        indice = i;
      }
    }

    return indice;
  }
}

void main() {

  List<Vehiculo> coches;
  int numVehiculos, cocheBarato;
  String marca, modelo;
  double precio;

  stdout.write("Digite el número de vehículos: ");
  numVehiculos = int.parse(stdin.readLineSync());

  coches = new List<Vehiculo>(numVehiculos);

  for (int i = 0; i < numVehiculos; i++) {
    print("Digite los datos del vehículo ${i + 1}:");
    stdout.write("Marca: ");
    marca = stdin.readLineSync();
    stdout.write("Modelo: ");
    modelo = stdin.readLineSync();
    stdout.write("Precio: ");
    precio = double.parse(stdin.readLineSync());

    coches[i] = new Vehiculo.datos(marca, modelo, precio);
  }

  cocheBarato = Vehiculo.indiceMasBarato(coches, numVehiculos);

  print("\nEl vehículo más barato es: ");
  coches[cocheBarato].mostrarDatos();
}